grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/distributed/Scheduler.java

```diff
@@ -157,33 +161,21 @@ public void close() {
 
     /**
      * Schedule a task to be submitted to the work queue when it is supposed to be run
-     * @param id id of the task to be scheduled
-     * @param configuration configuration of task to be scheduled, will be copied to WORK_QUEUE_TOPIC
-     */
-    private void scheduleTask(String id, String configuration) {
-        TaskState state = storage.getState(id);
-        scheduleTask(id, configuration, state);
-    }
-
-    /**
-     * Schedule a task to be submitted to the work queue when it is supposed to be run
-     * @param id id of the task to be scheduled
-     * @param configuration configuration of task to be scheduled, will be copied to WORK_QUEUE_TOPIC
      * @param state state of the task
      */
-    private void scheduleTask(String id,  String configuration, TaskState state) {
+    private void scheduleTask(TaskState state) {
         long delay = Duration.between(Instant.now(), state.runAt()).toMillis();
 
         markAsScheduled(state);
         if(state.isRecurring()) {
             Runnable submit = () -> {
                 markAsScheduled(state);
-                sendToWorkQueue(id, configuration);
+                sendToWorkQueue(state);
```

Comment:
And re: second question. 
If we first mark it as scheduled first, its possible we could crash before sending it to the work queue. This means it will remain in the new task queue and be picked up on the next scheduler restart.
If we mark it as scheduled after submitting, its possible the task would be picked up by the `TaskRunner` before being marked as `SCHEDULED`. At the moment, the `TaskRunner` will not run any tasks it receives that are not marked as `SCHEDULED`, so that task would never be run.